LoopKitUI/Views/Settings Editors/TherapySettingsViewModel.swift

```diff
@@ -40,6 +40,7 @@ public class TherapySettingsViewModel: ObservableObject {
                 chartColors: ChartColorPalette,
                 didSave: SaveCompletion? = nil) {
         self.mode = mode
+        let therapySettings = therapySettings()
```

Comment:
If you're only calling the therapy settings provider func once on init, isn't that the same as passing in the actual value (rather than a func) on init? This seems like a no-op.

Reply:
The bug did not occur whilst in `TherapySettingsView`... it was posted as a problem while in `SettingsView`.  So this is re-evaluated upon re-entry.

Reply:
The last step of the steps to reproduce says "Tap on Therapy Settings, and it will show the old settings without any updates from step 2.", so the staleness problem does surface in Therapy Settings, but I believe what you might be saying is that "SettingsViewModel" is caching it, which I think is a correct statement.

But I still don't understand how this change in itself changes anything. If, at the time of `TherapySettingsViewModel` instantiation, the `therapySettings()` evaluation is correct, then can't the caller just pass in the actual TherapySettings struct as before?

Reply:
This pattern is used in a bunch of other places: passing in a function to give the updated value (see also `pumpSupportedIncrements`, `syncPumpSchedule`).  What is the nature of your concern?

Reply:
In places where it's needed, that's fine, but it's not actually used here, beyond the constructor, which essentially exposes the exact same functionality as passing the struct directly. If a more complicated type (functor) is not needed, then the simpler struct should be used.

This also does not fix the issue, as it does not change behavior in any way; the issue must've been fixed elsewhere.

Reply:
@ps2 ok I will revert this